commit 8313fa0f1ca277e9633a78f461804abc3c5515b8
Author: Mark Thomas <markt@apache.org>
Date:   Thu Sep 17 09:26:08 2015 +0000

    Fix https://bz.apache.org/bugzilla/show_bug.cgi?id=58392
    Double-checked locking needs to use volatile to be thread-safe
    
    git-svn-id: https://svn.apache.org/repos/asf/tomcat/trunk@1703545 13f79535-47bb-0310-9956-ffa450edef68

diff --git a/java/org/apache/catalina/tribes/group/interceptors/DomainFilterInterceptor.java b/java/org/apache/catalina/tribes/group/interceptors/DomainFilterInterceptor.java
index 34672e3..f83a900 100644
--- a/java/org/apache/catalina/tribes/group/interceptors/DomainFilterInterceptor.java
+++ b/java/org/apache/catalina/tribes/group/interceptors/DomainFilterInterceptor.java
@@ -37,7 +37,7 @@ import org.apache.juli.logging.LogFactory;
 public class DomainFilterInterceptor extends ChannelInterceptorBase {
     private static final Log log = LogFactory.getLog(DomainFilterInterceptor.class);
     protected static final StringManager sm = StringManager.getManager(DomainFilterInterceptor.class);
-    protected Membership membership = null;
+    protected volatile Membership membership = null;
 
     protected byte[] domain = new byte[0];
 
